endpoints/openrtb2/video_auction.go

```diff
@@ -198,6 +201,12 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 			labels.CookieFlag = pbsmetrics.CookieFlagYes
 		}
 	}
+	if labels.PubID == "" && bidReq.Site != nil && bidReq.Site.Publisher != nil {
```

Comment:
I think we can get away with skipping the `labels.PubID == ""` check in this if statement because the next if statement on L207 checks that and assigns `labels.PubID` to `UNKNOWN` if true.

Reply:
Haven't noticed that. The logic is valid though because in this type of request we'll either have an `App` or a `Site`, therefore, if `labels.PubID == ""` is true, we can only find the `Publisher.ID` in the `Site` field or it is just `UNKNOWN`
```
 || labels.PubID == "" | bidReq.Site != nil | bidReq.Site.Publisher != nil  ||      A & B & C
1||     true           |      true          |         true                  ||  labels.PubID = bidReq.Site.Publisher.ID
2||     true           |      true          |         false                 ||  labels.PubID = "UNKNOWN"
3||     true           |      false         |         true                  ||  labels.PubID = "UNKNOWN"
4||     true           |      false         |         false                 ||  labels.PubID = "UNKNOWN"
5||     false          |      true          |         true                  ||  labels.PubID = bidReq.App.Publisher.ID
6||     false          |      true          |         false                 ||  labels.PubID = bidReq.App.Publisher.ID
7||     false          |      false         |         true                  ||  labels.PubID = bidReq.App.Publisher.ID
8||     false          |      false         |         false                 ||  labels.PubID = bidReq.App.Publisher.ID
```

Reply:
Tell you what, in order for the code to look cleaner, maybe the following rearrangement looks better:
```
	if labels.PubID == "" {
	    if bidReq.Site != nil && bidReq.Site.Publisher != nil {
            labels.PubID = bidReq.Site.Publisher.ID
        } else {
		    labels.PubID = "UNKNOWN"
        }
	}
```
Do you agree with this approach?

Reply:
I think I misread this. You're right. It's fine as is. Sorry about that!

Reply:
The proposed change above is also slightly more efficient than the code currently in review:  it avoids repeating the check of labels.PubID.

Reply:
Done